RFS/src/main/java/com/rfs/common/http/ConnectionContext.java

```diff
@@ -85,9 +92,24 @@ public static interface Params {
 
         String getPassword();
 
+        String getSigv4Region();
+
+        String getSigv4ServiceName();
+
         boolean isInsecure();
     }
 
+    @Builder
+    @Getter
+    public static class ConstParams implements Params {
+        private String host;
+        private String username;
+        private String password;
+        private String sigv4Region;
+        private String sigv4ServiceName;
+        private boolean insecure;
+    }
+
     @Getter
     public static class TargetArgs implements Params {
```

Comment:
Where is the role assumed for the request passed in?

Reply:
When creating a ConnectionContext, we use `DefaultCredentialsProvider.create()` which is consistent with the Replayer and RFS S3 Access https://github.com/opensearch-project/opensearch-migrations/blob/7ab6fc46c983b71bf190f43f9948ee3693034ca0/RFS/src/main/java/com/rfs/common/S3Repo.java#L91

If we think we need the feature to use credentials to assume a different role, let's capture that in a JIRA to update it throughout the solution.

Reply:
https://opensearch.atlassian.net/browse/MIGRATIONS-1895